compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -685,6 +685,7 @@ object Trees {
   case class SplicedExpr[+T <: Untyped] private[ast] (expr: Tree[T], tpt: Tree[T])(implicit @constructorOnly src: SourceFile)
     extends TermTree[T] {
     type ThisTree[+T <: Untyped] = SplicedExpr[T]
+    def isInBraces: Boolean = span.end != expr.span.end
```

Comment:
Looks like this isn't new, but it looks a bit fragile: what happens if the Splice node is compiler/macro-generated and doesn't have a position different from the expr?

Reply:
The only place where we use this method is when typing splice that comes from the parser. Therefore the current usage should be correct in all cases. But having this method available could lead to someone using it where they should not. I will move this logic into the quote typing logic.